server/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -629,6 +630,7 @@ boolean isIncomplete() {
 
             stopperRunner.accept("http-server-transport-stop", injector.getInstance(HttpServerTransport.class)::close);
             stopperRunner.accept("async-search-stop", () -> awaitSearchTasksComplete(maxTimeout));
+            stopperRunner.accept("reindex-stop", () -> awaitReindexTasksComplete(TimeValue.min(TimeValue.timeValueMinutes(1), maxTimeout)));
```

Comment:
`maxTimeout=ZERO` seems to mean infinite, I wonder if the `TimeValue.min` here should respect that?